allauth/socialaccount/providers/keap/views.py

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+import requests
+
+from allauth.socialaccount import app_settings
+from allauth.socialaccount.providers.keap.provider import KeapProvider
+from allauth.socialaccount.providers.oauth2.views import (
+    OAuth2Adapter,
+    OAuth2CallbackView,
+    OAuth2LoginView,
+)
+
+
+class KeapOAuth2Adapter(OAuth2Adapter):
+    provider_id = KeapProvider.id
+    supports_state = True
+
+    access_token_url = "https://api.infusionsoft.com/token"
+    authorize_url = "https://accounts.infusionsoft.com/app/oauth/authorize"
+    profile_url = "https://api.infusionsoft.com/crm/rest/v1/oauth/connect/userinfo"
+
+    def complete_login(self, request, app, token, **kwargs):
+        response = kwargs.get("response", None)
+        scope = response.get("scope")
+        app_name = scope.split("|")[1].split(".")[0]
+
+        extra_data = requests.get(
```

Comment:
Can you split this up in something like?:

```
response = requests.get(...)
response.raise_for_status()
extra_data = response.json()
```

Reply:
@pennersr Done, let me know if I should address anything else. Thanks for taking a look!

Reply:
If you can have a look at the test suite -- it is currently failing.

Reply:
Ok, I think I have gotten what was needed. Sorry for the back and forth, pretty new to testing/contributing.